galata/src/helpers/notebook.ts

```diff
@@ -1255,14 +1255,37 @@ export class NotebookHelper {
 
     await this._setCellMode(cell, 'Edit');
     await cell.getByRole('textbox').press('Control+A');
-    await cell
-      .getByRole('textbox')
-      .type(source, { delay: cellType === 'code' ? 100 : 0 });
+    await cell.getByRole('textbox').type(source, { delay: 0 });
     await this._setCellMode(cell, 'Command');
 
-    // give CodeMirror time to style properly
     if (cellType === 'code') {
-      await this.page.waitForTimeout(500);
+      // Wait until the CodeMirror highlighting is stable
+      // over 10 consecutive animation frames.
+      await cell.evaluate((cell: HTMLElement) => {
```

Comment:
Do you think `toHaveScreenshot` will work in this case - trying to simplify the code if possible?

Ref: https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-2

Reply:
I think the problem with most playwright's "is it stable" logic is that these only look at 2 consecutive frames:

> toHaveScreenshot(): This function will wait until two consecutive locator screenshots yield the same result, and then compare the last screenshot with the expectation.

but CodeMirror takes rather long pauses when rendering so I am not actually sure if even 10 animation frames is enough for some cases.